src/storage/cache/rpc/message.rs

```diff
@@ -0,0 +1,214 @@
+use bytes::{BufMut, BytesMut};
+
+use crate::async_fuse::util::usize_to_u64;
+
+use super::{
+    error::RpcError,
+    packet::{Decode, Encode},
+    utils::get_u64_from_buf,
+};
+
+/// The request type of the request.
+#[derive(Debug)]
+pub enum ReqType {
+    /// The keep alive request.
+    KeepAliveRequest,
+    /// The file block request.
+    FileBlockRequest,
+}
+
+impl ReqType {
+    /// Convert u8 to `ReqType`
+    pub fn from_u8(op: u8) -> Result<Self, RpcError> {
+        match op {
+            0 => Ok(Self::KeepAliveRequest),
+            1 => Ok(Self::FileBlockRequest),
+            _ => Err(RpcError::InternalError(format!(
+                "Invalid operation type: {op}"
+            ))),
+        }
+    }
+
+    /// Convert `ReqType` to u8
+    #[must_use]
+    pub fn to_u8(&self) -> u8 {
+        match *self {
+            Self::KeepAliveRequest => 0,
+            Self::FileBlockRequest => 1,
+        }
+    }
+}
+
+/// The operation type of the response.
+#[derive(Debug)]
+pub enum RespType {
+    /// The keep alive response.
+    KeepAliveResponse,
+    /// The file block response.
+    FileBlockResponse,
+}
+
+impl RespType {
+    /// Convert u8 to `RespType`
+    pub fn from_u8(op: u8) -> Result<Self, RpcError> {
+        match op {
+            0 => Ok(Self::KeepAliveResponse),
+            1 => Ok(Self::FileBlockResponse),
+            _ => Err(RpcError::InternalError(format!(
+                "Invalid operation type: {op}"
+            ))),
+        }
+    }
+
+    /// Convert `RespType` to u8
+    #[must_use]
+    pub fn to_u8(&self) -> u8 {
+        match *self {
+            Self::KeepAliveResponse => 0,
+            Self::FileBlockResponse => 1,
+        }
+    }
+}
+
+/// Common data structures shared between the client and server.
+#[derive(Debug, Default, Clone)]
+pub struct FileBlockRequest {
+    /// The file ID.
+    pub file_id: u64,
+    /// The block ID.
+    pub block_id: u64,
+    /// The block size.
+    pub block_size: u64,
+}
+
+impl Encode for FileBlockRequest {
+    /// Encode the file block request into a byte buffer.
+    fn encode(&self, buf: &mut BytesMut) {
+        encode_file_block_request(self, buf);
+    }
+}
+
+impl Decode for FileBlockRequest {
+    /// Decode the byte buffer into a file block request.
+    fn decode(buf: &[u8]) -> Result<Self, RpcError> {
+        decode_file_block_request(buf)
+    }
+}
+
+/// Decode the file block request from the buffer.
+pub fn decode_file_block_request(buf: &[u8]) -> Result<FileBlockRequest, RpcError> {
```

Comment:
The decode & encode better be member function.
Should add some test for encode and then decode to valid the behavior

Reply:
Ok, I have add the unit tests.